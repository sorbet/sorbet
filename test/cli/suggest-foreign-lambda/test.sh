#!/bin/bash

tmp="$(mktemp -d)"
infile="test/cli/suggest-foreign-lambda/suggest-foreign-lambda.rb"
cp "$infile" "$tmp"

cwd="$(pwd)"
cd "$tmp" || exit 1

"$cwd/main/sorbet" --silence-dev-message -a suggest-foreign-lambda.rb 2>&1

echo
echo --------------------------------------------------------------------------
echo

# Also cat the file, to make sure that 'extend' is only added once per class.
cat suggest-foreign-lambda.rb

rm suggest-foreign-lambda.rb
rm -r "$tmp"
