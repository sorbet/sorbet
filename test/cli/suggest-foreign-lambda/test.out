suggest-foreign-lambda.rb:7: The argument to `foreign:` must be a lambda https://srb.help/3508
     7 |  prop :my_model, String, foreign: MyModel
                                           ^^^^^^^
  Autocorrect: Done
    suggest-foreign-lambda.rb:7: Replaced with `-> {MyModel}`
     7 |  prop :my_model, String, foreign: MyModel
                                           ^^^^^^^
Errors: 1

--------------------------------------------------------------------------

# typed: strict

class MyModel; end

class MyOtherModel
  include T::Props
  prop :my_model, String, foreign: -> {MyModel}
end
